src/compiler/program.ts

```diff
@@ -3347,9 +3343,6 @@ namespace ts {
         return resolveConfigFileProjectName(passedInRef.path);
     }
 
-    function getEmitDeclarationOptionName(options: CompilerOptions) {

```

Comment:
You want to change `getProjectReferenceRedirectProject` to ensure we are redirecting to d.ts files of js files from the composite projects (so a test case for same)